tasks/setup.yml

```diff
@@ -12,40 +12,69 @@
     state: directory
     recurse: yes
 
+- name: Check steam is already where it should be
+  stat:
+    path: "{{ steamcmd_directory }}/steamcmd.sh"
+  register: stat_steamcmd_result
+
 - name: download SteamCMD for Linux
   get_url:
     url: https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz
     dest: "{{ steamcmd_directory }}steamcmd_linux.tar.gz"
-    force: yes
-    headers: "User-Agent: lutangar/ansible-role-steamcmd"
-    validate_certs: no
+  when: stat_steamcmd_result.stat.exists == False
 
 - name: extract the contents to cmd directory
   unarchive:
     src: "{{ steamcmd_directory }}steamcmd_linux.tar.gz"
     dest: "{{ steamcmd_directory }}"
     copy: no
+  when: stat_steamcmd_result.stat.exists == False
+
+- name: attempt to log into steam anonymously
```

Comment:
Nice catch :+1: 
Whereas I understand the workaround, I would prefer something like the following based on `block` should work:
https://github.com/ansible/ansible/issues/4297#issuecomment-356427588
For what I can recall, there should be a way to tackle this with a conditionnal include of another set of tasks.

Reply:
block also does not work. Even if the block is conditional and skipped, the register inside still gets the skipped info. see https://github.com/ansible/ansible/issues/4297#issuecomment-386962997
I think the set_fact works fine here, but if you have another solution I will test it.